indexing-service/src/main/java/org/apache/druid/indexing/scheduledbatch/ScheduledBatchTaskManager.java

```diff
@@ -134,72 +144,101 @@ public void startScheduledIngestion(
   public void stopScheduledIngestion(final String supervisorId)
   {
     log.info("Stopping scheduled batch ingestion for supervisorId[%s].", supervisorId);
-    final SchedulerManager manager = supervisorToManager.get(supervisorId);
-    if (manager != null) {
+    final ScheduledBatchTask taskScheduler = supervisorToTaskScheduler.get(supervisorId);
+    if (taskScheduler != null) {
       // Don't remove the supervisorId from supervisorToManager. We want to be able to track the status
       // for suspended supervisors to track any in-flight tasks. stop() will clean up all state completely.
-      manager.stopScheduling();
+      taskScheduler.stopScheduling();
     }
   }
 
+  /**
+   * Starts the scheduled batch task manager by registering the {@link TaskRunnerListener}.
+   * This allows tracking of any tasks submitted by the batch supervisor.
+   * <p>
+   * Should be invoked when the Overlord service starts or during leadership transitions.
+   * </p>
+   */
   public void start()
   {
-    log.info("Starting scheduled batch scheduler.");
+    log.info("Starting scheduled batch task manager.");
     final Optional<TaskRunner> taskRunnerOptional = taskMaster.getTaskRunner();
     if (taskRunnerOptional.isPresent()) {
       taskRunnerOptional.get().registerListener(taskRunnerListener, Execs.directExecutor());
     } else {
-      log.warn("Task runner not registered for scheduled batch scheduler.");
+      log.warn("Task runner not registered for scheduled batch task manager.");
     }
   }
 
+  /**
+   * Stops the scheduled batch task manager by unregistering the previously registered {@link TaskRunnerListener}.
+   * <p>
+   * Should be invoked when the Overlord service stops or during leadership transitions.
+   * </p>
+   */
   public void stop()
   {
-    log.info("Stopping scheduled batch scheduler.");
+    log.info("Stopping scheduled batch task manager.");
     final Optional<TaskRunner> taskRunnerOptional = taskMaster.getTaskRunner();
     if (taskRunnerOptional.isPresent()) {
       taskRunnerOptional.get().unregisterListener(taskRunnerListener.getListenerId());
     }
 
-    supervisorToManager.clear();
+    supervisorToTaskScheduler.clear();
   }
 
   @Nullable
-  public ScheduledBatchSupervisorSnapshot getSchedulerSnapshot(final String supervisorId)
+  public ScheduledBatchSupervisorStatus getTaskManagerStatus(final String supervisorId)
   {
-    final SchedulerManager manager = supervisorToManager.get(supervisorId);
+    final ScheduledBatchTask manager = supervisorToTaskScheduler.get(supervisorId);
     if (manager == null) {
       return null;
     }
 
-    final ScheduledBatchStatusTracker.BatchSupervisorTaskStatus tasks = statusTracker.getSupervisorTasks(supervisorId);
-    return new ScheduledBatchSupervisorSnapshot(
+    final BatchSupervisorTaskReport taskStatus = statusTracker.getSupervisorTaskStatus(supervisorId);
+    return new ScheduledBatchSupervisorStatus(
```

Comment:
Rather than duplicating fields between `BatchSupervisorTaskReport` and `ScheduledBatchSupervisorStatus`, just add a field `BatchSupervisorTaskReport taskReport` in `ScheduledBatchSupervisorStatus`.